### PR TITLE
Fixed grid hook calls

### PR DESCRIPTION
**Pull request by mickaelandrieu**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.5.x
| Description?  | Regarding the docs on available Grid hooks, dispatched hooks name are wrong **and** A hook is missing
| Type?         | bug fix
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | none, but will happen soon.
| How to test?  | Create a module that use all hooks of a Grid and check they are propagated. **This means Grid hooks are not really available**

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/12389)

[customers.zip](https://github.com/PrestaShop/PrestaShop/files/2832660/customers.zip)

### CODE REVIEW

**Comment by mickaelandrieu** [comment]
ping @eternoendless: this must be merged in 1.7.5.x, because Grid component is not really available otherwise :/

File: src/Core/Grid/GridFactory.php [hunk 1]
```diff
@@ -74,6 +78,10 @@ public function getGrid(SearchCriteriaInterface $searchCriteria)
         $definition = $this->definitionFactory->getDefinition();
         $data = $this->dataFactory->getData($searchCriteria);
 
+        $this->hookDispatcher->dispatchWithParameters('action' . Container::camelize($definition->getId()) . 'GridDataModifier', [
+            'data' => $data,
```
Comment (sarjon) [inline src/Core/Grid/GridFactory.php]: `$data` is immutable, isn't it?

Comment (PierreRambaud) [reply src/Core/Grid/GridFactory.php]: It should not and be passed by reference. No?

Comment (mickaelandrieu) [reply src/Core/Grid/GridFactory.php]: make sense @PierreRambaud , ping @sarjon would you mind to complete my pull request, making $data a reference and not a variable?

**Review by matks — APPROVED** [review]

**Comment by marionf** [comment]
@mickaelandrieu Do you have a module allowing me to test your PR ? :)

**Comment by mickaelandrieu** [comment]
@marionf sure, in the pull request description :+1:

**Comment by matks** [comment]
Thanks @mickaelandrieu

**Pull request merged**